packages/core/lib/console.js

```diff
@@ -61,22 +61,31 @@ class Console extends EventEmitter {
 
   async start() {
     try {
-      // we calculate the context before calling `start`  so that the context
-      // variables are available as early as possible after the repl starts
-      const context = { ...(await this.calculateTruffleAndUserGlobals()) };
-      // we set up this.repl here because `this.provision` sets values on
-      // this.repl.context - this will finish setting up the environment
-      this.repl = { context };
-      // this hydrates the environment with the user's contracts
-      this.provision();
-
+      // start the repl with an empty prompt and show a proper one when
+      // the repl has set up its context and is ready to accept input
       this.repl = repl.start({
-        prompt: "truffle(" + this.options.network + ")> ",
+        prompt: "",
+        ignoreUndefined: true,
         eval: this.interpret.bind(this)
       });
-      this.repl.context = context;
 
-      //want repl to exit when it receives an exit command
+      // Get and set Truffle and User Globals
+      const truffleAndUserGlobals = await this.calculateTruffleAndUserGlobals() ;
+      Object.entries(truffleAndUserGlobals).forEach(([key, value]) => {
+        Object.defineProperty(this.repl.context, key, {
+          configurable: false,
+          enumerable: true,
+          value,
+        });
```

Comment:
This is a breaking change that we may not want. Users will no longer be able to delete truffleAndUserGlobals from the global context. This means, accounts for example:
```
delete this.accounts // will not delete or reassign accounts
accounts.splice(1,5) // will still change the contents, along with push, pop
```

Reply:
After discussing with @eggplantzzz we feel users should have unfettered access to the context.